experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -78,3 +107,356 @@ describe("lazyField", () => {
 		);
 	});
 });
+
+/**
+ * Initializes a test tree, context, and cursor, and moves the cursor to the tree's root.
+ *
+ * @returns The initialized context and cursor.
+ */
+function initializeTreeWithContent<Kind extends FieldKind, Types extends AllowedTypes>(
+	treeContent: TreeContent,
+): {
+	context: Context;
+	cursor: ITreeSubscriptionCursor;
+} {
+	const context = contextWithContentReadonly(treeContent);
+	const cursor = initializeCursor(context, rootFieldAnchor);
+
+	return {
+		context,
+		cursor,
+	};
+}
+
+describe("LazyOptionalField", () => {
+	describe("as", () => {
+		it("Any", () => {
+			// #region Tree and schema initialization
+
+			const builder = new SchemaBuilder("test");
+			const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
+			const recursiveStructSchema = builder.structRecursive("recursiveStruct", {
+				flag: SchemaBuilder.fieldValue(booleanLeafSchema),
+				child: SchemaBuilder.fieldRecursive(
+					FieldKinds.optional,
+					() => recursiveStructSchema,
+				),
+			});
+			const rootSchema = SchemaBuilder.fieldOptional(builder.struct("struct", {}));
+			const schema = builder.intoDocumentSchema(rootSchema);
+
+			const { context, cursor } = initializeTreeWithContent({ schema, initialTree: {} });
+
+			// #endregion
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(Any),
+				cursor,
+				detachedFieldAnchor,
+			);
+
+			// Positive cases
+			assert(field.is(SchemaBuilder.fieldOptional(Any)));
+
+			// Negative cases
+			assert(!field.is(SchemaBuilder.fieldOptional()));
+			assert(!field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
+			assert(!field.is(SchemaBuilder.fieldValue(Any)));
+			assert(!field.is(SchemaBuilder.fieldSequence(Any)));
+			assert(
+				!field.is(SchemaBuilder.fieldRecursive(FieldKinds.value, recursiveStructSchema)),
+			);
+		});
+
+		it("Primitive", () => {
+			// #region Tree and schema initialization
+
+			const builder = new SchemaBuilder("test");
+			const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
+			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+			const recursiveStructSchema = builder.structRecursive("recursiveStruct", {
+				flag: SchemaBuilder.fieldValue(booleanLeafSchema),
+				child: SchemaBuilder.fieldRecursive(
+					FieldKinds.optional,
+					() => recursiveStructSchema,
+				),
+			});
+			const rootSchema = SchemaBuilder.fieldOptional(builder.struct("struct", {}));
+			const schema = builder.intoDocumentSchema(rootSchema);
+
+			const { context, cursor } = initializeTreeWithContent({ schema, initialTree: {} });
+
+			// #endregion
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(booleanLeafSchema),
+				cursor,
+				detachedFieldAnchor,
+			);
+
+			// Positive cases
+			assert(field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
```

Comment:
Do we want to also check that `fieldValue()` works? It'd be a pain to write another whole test just to check it, but what if you create a helper function which takes (or makes) a schema builder and then creates `booleanLeafSchema`, `numberLeafSchema`, `recursiveLeafSchema`, etc. and returns them all? Then each of these tests would only need to call the helper and set the root schema for the document. It would remove a lot of duplicated code (since many of these test cases use similar or the same kinds of schema) and would make it easier to add new test cases. It's okay if a test doesn't use everything in the schema.

I assume we'll be adding more tests, so totally fine if you want to get this checked in as a baseline and factor stuff out later.

Reply:
That's a good idea. I think I will wait and refactor when adding future tests, though.